### DISCUSSION TITLE
[Question]: How to create arbitrary events via MQTT

### DISCUSSION THREAD

**Discussion by DD81** [blakeblackshear/frigate#12341] [Ask A Question] [answered]
### What is your question:

I use Frigate with HomeAssistant integration and record constantly.

I have set up an automation that moves the PTZ to a specific preset when someone enters a designated zone on a fixed camera.
I want to create Events whenever this automation is activated.

Is there a way to force the creation of Events for a specific camera via MQTT?

**Comment by NickM-27** [comment, marked as answer]
this can not be done with mqtt, it must be done with the http api

**Comment by DD81** [comment]
Thanks for your support.
I did not know that the HTTP API had such a feature...
It looks like the following POST method will accomplish my goal.

POST /api/events/<camera_name>/<label>/create

Thank you very much.

**Comment by DD81** [comment]
I was able to force the HTTP API to create Events.
However, there is no audio.
I think it is because my PTZ camera does not have a microphone and I am duplicating the audio from a nearby fixed camera using "ffmpeg:cam_name#audio=copy".
Is there any way to create Events with audio in this situation?

**Reply by NickM-27** [reply]
the creation of the manual events has nothing to do with audio, if the camera supports audio then the recordings will have audio

**Reply by DD81** [reply]
If I want to record audio when manually creating an event, does the camera need to support direct audio?
I have the following setup but the event created by object detection records audio duplicated from c1.
Why is there no audio when I create an event via HTTP API?

go2rtc:
  streams:
    c1:
      - rtsp://user:pass@camera-ip:554/cam/realmonitor?channel=1&subtype=0
    ptz1:
      - rtsp://user:pass@ptz-ip:554/cam/realmonitor?channel=1&subtype=0
      - "ffmpeg:c1#audio=copy"
cameras:
  c1:
    enabled: True
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/c1
          input_args: preset-rtsp-restream
          roles:
            - record
            - audio
            - detect
  ptz1:
    enabled: True
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/ptz1
          input_args: preset-rtsp-restream
          roles:
            - record
            - audio
            - detect

**Reply by NickM-27** [reply]
all creating an event does is tell the recorded the same thing that a standard event does, there is no difference in how this is handled. So I don't see any reason one would have audio and another wouldn't
